pkg/irsa/capa/capa.go

```diff
@@ -95,17 +95,13 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		return err
 	}
 
-	// Fetch custom tags from Cluster CR
-	cluster := &capi.Cluster{}
-	err = s.Client.Get(ctx, types.NamespacedName{Namespace: s.Scope.ClusterNamespace(), Name: s.Scope.ClusterName()}, cluster)
-	if apierrors.IsNotFound(err) {
-		// fallthrough
-	} else if err != nil {
+	// Fetch custom tags from AWSCluster CR
+	awsCluster := &capa.AWSCluster{}
+	err = s.Client.Get(ctx, types.NamespacedName{Namespace: s.Scope.ClusterNamespace(), Name: s.Scope.ClusterName()}, awsCluster)
+	if err != nil {
 		return err
```

Comment:
Should we specifically handle `apierrors.IsNotFound(err)`, as before?

Reply:
I do not think its a good idea to continue if the AWSCluster CR is missing, because in that case either something terrible happened or it's some kind of super early in the creation processes and it would mean we ignore the tags that are supposed to be there.


this code feels like it was copied from somewhere where it make sense

Reply:
I removed it on purpose